minicluster/src/main/java/alluxio/multi/process/PortCoordination.java

```diff
@@ -33,14 +33,11 @@
       // add ports here to avoid conflicting with other processes on those ports.
   ));
 
-  public static final List<ReservedPort> EMBEDDED_JOURNAL_FAILOVER = allocate(3, 0);
-  public static final List<ReservedPort> EMBEDDED_JOURNAL_RESTART = allocate(3, 0);
-  public static final List<ReservedPort> EMBEDDED_JOURNAL_RESTART_STRESS = allocate(3, 0);
```

Comment:
what do these tests use now?

Reply:
@ZacBlanco Is one list necessary per each test, or can tests grouped under the same class reuse the same ports? Also, could the `allocate` method be made public so that tests can dynamically requests new ports at runtime, or would that cause collisions? Thanks in advance.

Reply:
Multiple lists is better because we've seen cases where tests that run sequentially and attempt to re-use the same port fail to start because the OS in the background hasn't actually fully closed the socket (usually due to some TCP timeout or high system load). It just prevents some flakiness in the test if we can allocate ports for each test individually, even if their execution doesn't overlap.

Dynamically allocating at runtime could lead to more potential problems if there are conflicts. AFAIK the port coordination uses filesystem locks (`man 2 flock`) to try to grab exclusive access to the ports, but the locks are simply advisory and aren't a strong guarantee.